File: Programmers/43165.py
from collections import deque


def solution(numbers, target):
    answer = 0

    queue = deque()
    queue.append((0, 0))
    while queue:
        s, l = queue.popleft()
        if s == target and l == len(numbers):
            answer += 1
        if l == len(numbers):
            continue
        queue.append((s + numbers[l], l + 1))
        queue.append((s - numbers[l], l + 1))

    return answer

File: Programmers/43165.js
// DFS,BFS 타겟 넘버
function solution(numbers, target) {
    let answer = 0;
    const len = numbers.length;

    function dfs(cnt, sum) {
        if (cnt === len) {
            if (sum === target)
                answer++;
            return;
        }

        dfs(cnt + 1, sum + numbers[cnt]);
        dfs(cnt + 1, sum - numbers[cnt]);
    }

    dfs(0, 0);

    return answer;
}